### ISSUE TITLE
Record not importing from iNaturalist

### ISSUE THREAD

**Issue by robin-hutchinson** [BiologicalRecordsCentre/iRecord#1585]
This record: https://www.inaturalist.org/observations/180600919 hasn't come through from iNaturalist (it is research grade and CC BY-NC).  The species has recently been added to the UKSI as it is new to the UK which might be the issue, but the other Carabus auronitens record from iNaturalist has been imported successfully.

**Comment by kitenetter** [comment]
The record is flagged as "introduced" - I can't remember if we are excluding such records @johnvanbreda ?

**Comment by DavidRoy** [comment]
we shouldn't be excluding introduced species imho

**Comment by johnvanbreda** [comment]
We don't exclude introduced species, just captive organisms.

At some point, iNat has added an additional sex term "Cannot be Determined" which we didn't have in our stage termlists. The records with this term were failing validation and being logged in a table that tracks sync failures. I've now given the iNat data it's own custom sex termlist and re-imported that record.

At some point we should do the following:
* Alter the iNat  import code so it can create new terms on the fly.
* Re-attempt importing all the records that have failed import previously. We should really do that after the taxon mappings table has been populated (see #1222).

**Comment by kitenetter** [comment]
Thanks John. I'll leave this open and as you say e can come back to it after the taxon dictionary stuff.

**Comment by Sam-Amy** [comment]
> We don't exclude introduced species, just captive organisms.

Does this refer to the iNat flag 'Captive/cultivated'? In meeting with BSBI and @sacrevert recently the suggestion to add a similar field i.e. 'not in wild' and link to this iNat field was raised.

**Comment by kitenetter** [comment]
@Sam-Amy yes I believe it does mean the Captive/cultivated tick box in iNat.

**Comment by johnvanbreda** [comment]
I've added code to automatically create terms in our sex, stage or vitality term list if new terms appear on the iNat side (develop branch). 

Still to do - trigger a refreshed import of records which failed due to the missing Cannot Be Determined term. This needs new code though we do have a log of the failed records in rest_api_sync_skipped_records.

**Comment by johnvanbreda** [comment]
Have now added code so once the mappings are uploaded into the `rest_api_sync_taxon_mappings` table, the previous failures can be rescanned. Info on how to do this in the rest_api_sync module example config file.

**Comment by kitenetter** [comment]
@johnvanbreda is there still work to do for this issue?

**Comment by johnvanbreda** [comment]
Yes, at some point we need to trigger an import of all the records that have been skipped due to errors. However it makes sense to do this once for both these issues and the taxon matching issues, since we will attempt to re-import all the skipped records irrespective of the reason. Trying to re-import skipped records at this point would result in most records failing again due to taxon matching.
